docs/client_reference.rst

```diff
@@ -2315,3 +2350,17 @@ Hierarchy of exceptions
     * :exc:`WSServerHandshakeError`
 
   * :exc:`InvalidURL`
+
+    * :exc:`InvalidUrlClientError`
+
+      * :exc:`InvalidUrlRedirectClientError`
+
+    * :exc:`NonHttpUrlClientError`
+
+      * :exc:`NonHttpUrlRedirectClientError`
```

Comment:
Per https://github.com/aio-libs/aiohttp/pull/6722#discussion_r1479985833 I maintain that these shouldn't be advertised as invalid. It sends a wrong message.

@Dreamsorcerer care to share an opinion?

Reply:
Yeah, I don't mind too much. I think some users will expect it your way and other users another way (i.e. InvalidUrl meaning any URL that can't be processed by aiohttp). Renaming InvalidURL, but keeping it as the parent could meet both expectations, but would break backwards-compatibility.

Reply:
I updated inheritance, it's not subclass of `ClientError`